docs/Makefile

```
@@ -8,10 +8,54 @@ SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _build
 
+JUPYTEXT      ?= jupytext
+NBCONVERT     ?= jupyter nbconvert
+DEMOFOLDER   = demos/demos
+OUTFOLDER    = demos/out
+
+# Script to add download buttons to the bottom of the rst files
+ADD_DOWNLOAD_BUTTONS ="$\
+\$$a$\
+:download:\`Download Python source code \<$$(basename $$1 .rst).py\>\`\n\n$\
+:download:\`Download as Jupyter Notebook \<$$(basename $$1 .rst).ipynb\>\`\n"
+
+# Script to add file names as labels to the top of the rst files
+ADD_FILE_NAME_LABEL = "1s/^/.. _$$(basename $$1 .rst):\n\n/"
```

You can also use sed's **insert command** to prepend text (slightly simpler expression). When combined with **parameter expansion** switch as suggested below:
```suggestion
ADD_FILE_NAME_LABEL = "1i .. _$${1%.rst}:\n"
```